label_errors.py

```diff
@@ -0,0 +1,187 @@
+)
+@click.option("--verbose", default=0, type=int, help="Verbose level")
+def main(
+    api_endpoint: str,
+    api_key: str,
+    cv_folds: int,
+    clear_dataset_cache: bool,
+    label_types: str,
+    max_assets: int,
+    model_name: str,
+    project_id: str,
+    training_epochs: int,
+    upload_errors: bool,
+    verbose: int,
+):
+    """
+    Main method for detecting incorrect labeled assets in a Kili project.
+    It downloads the assets, trains a classification neural network with CV and then by
+    using the Cleanlab library we get the wrong labels. The concerned asset IDs are then
+    stored in a file, but also a metadata (labeling_error: true) is uploaded to Kili to
+    easily filter them later in the app.
+    """
+
+    kili = Kili(api_key=api_key, api_endpoint=api_endpoint)
+    input_type, jobs, _ = get_project(kili, project_id)
+
+    for job_name, job in jobs.items():
+        content_input = job.get("content", {}).get("input")
+        ml_task = job.get("mlTask")
+        if (
+            content_input == ContentInput.Radio
+            and input_type == InputType.Image
+            and ml_task == MLTask.Classification
+        ):
+            job_path = build_model_repository_path(HOME, project_id, job_name, "")
+            data_path = os.path.join(job_path, "data")
+            model_path = os.path.join(job_path, "model")
+
+            if clear_dataset_cache and os.path.exists(job_path) and os.path.isdir(job_path):
+                kili_print("Dataset cache for this project is being cleared.")
+                shutil.rmtree(job_path)
+
+            os.makedirs(job_path, exist_ok=True)
+            os.makedirs(data_path, exist_ok=True)
+            os.makedirs(model_path, exist_ok=True)
+
+            kili_print("Downloading datasets from Kili")
+            assets = get_assets(kili, project_id, parse_label_types(label_types), max_assets)
+            if len(assets) == 0:
+                raise Exception("No asset in dataset, exiting...")
+
+            download_assets(assets, api_key, data_path, job_name)
+
+            model_name = set_default(
+                model_name,
+                ModelName.EfficientNetB0,
+                "model_name",
+                [ModelName.EfficientNetB0, ModelName.Resnet50],
+            )
+
+            found_errors = train_and_get_error_labels(
+                cv_n_folds=cv_folds,
+                data_dir=data_path,
+                model_dir=model_path,
+                model_name=model_name,
+                training_epochs=training_epochs,
+                verbose=verbose,
+            )
+
+            print()
+            kili_print("Number of wrong labels found: ", len(found_errors))
+
+            if found_errors:
+                save_errors(found_errors, job_path)
+                if upload_errors:
+                    upload_errors_to_kili(found_errors, kili)
+        else:
+            kili_print("not implemented yet")
```

Comment:
`raise NotImplementedError`

Reply:
I removed the `kili_print()`, not sure if I had to replace it with the `raise` or just add the `raise` after